### PR TITLE
Extend subscription rhp with ability to subscribe by registry entry id

### PR DESCRIPTION
**Pull request by ChrisSchinnerl**
This MR adds another type of request to the subscription rhp to allow for subscribing only via registry ID.
The result is the same as for regular subscriptions except for the initial values we send to the client after subscribing. These now contain the public key to allow the client to verify that the responses belong to the values they requested.

### CODE REVIEW

File: modules/host/rpcsubscribe.go [hunk 1]
```diff
@@ -193,6 +232,39 @@ func (h *Host) managedHandleUnsubscribeRequest(info *subscriptionInfo, pt *modul
 	return errors.AddContext(err, "failed to signal successfully unsubscribing from entries")
```
Comment (n8maninger) [inline modules/host/rpcsubscribe.go:232]: This should also use `managedHandleFinalizeUnsubscribeRequest`

**Review by n8maninger — COMMENTED** [review]
Two small things we could address in a follow-up. @lukechampine, should we merge this and get an RC1 out this afternoon?

File: modules/host/rpcsubscribe.go (hunk 1 above)
Comment (ChrisSchinnerl) [reply modules/host/rpcsubscribe.go]: done

File: modules/renter/workersubscription_test.go [hunk 2]
```diff
@@ -1298,3 +1298,122 @@ func TestThreadedSubscriptionLoop(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// TestSubscribeUnsubsribeByRID tests the subscription helper methods against
+// the worker tester. They are already unit-tested against a host in
+// rpcsubscribe_test.go but better safe than sorry.
+func TestSubscribeUnsubsribeByRID(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (ChrisSchinnerl) [reply modules/renter/workersubscription_test.go]: and done

**Comment by lukechampine** [comment]
TestNewProgramLowCollateralBudget is failing -- any connection to this?

**Comment by n8maninger** [comment]
> 

Looks like another NDF

**Pull request merged**